spec.bs

```diff
@@ -798,10 +798,15 @@ input {{SecurePaymentConfirmationRequest}} |data|, are:
 1. For each |id| in |data|["{{SecurePaymentConfirmationRequest/credentialIds}}"]:
 
     1. Run the [=steps to silently determine if a credential is available for
-        the current device=] and the [=steps to silently determine if a
+        the current device=], passing in
+        |data|["{{SecurePaymentConfirmationRequest/rpId}}"] and |id|.
+        If the result is `false`, remove |id| from
+        |data|["{{SecurePaymentConfirmationRequest/credentialIds}}"].
+    1.  If the |data|["{{SecurePaymentConfirmationRequest/rpId}}"] is
+        not in the current origin, run the [=steps to silently determine if a
```

Comment:
Yeah, so we don't just say things like "is not in the current origin" in specs 🤣 . We try to make them a bit more like code. I think this should be:

At the top of this method, change to:

```
The [=steps to validate payment method data=] for this payment method, for an
input {{PaymentRequest}} |request| and {{SecurePaymentConfirmationRequest}} |data|, are:
```
(adding 'request', to get it in scope for looking up. This is a bit hand-wavy as Payment Request doesn't currently pass it in to the 'Steps to check if a payment can be made', but we can fix that later, not a big deal I think)

Then:

```
    1.  If the |data|["{{SecurePaymentConfirmationRequest/rpId}}"] is
        not the [=origin=] of the [=relevant settings object=] of |request|,
        run the [=steps to silently determine if a
```

Reply:
Definitely not :). What we are trying to do here is determine if the origin that is **calling SPC** is the RP's origin, and thus should be able to skip the check for the thirdPartyPayment bit.

For example, consider the following nested case:

```
<-- main frame is merchant.example -->
  <iframe src="psp.example/pay">
    <iframe src="issuer.example/3ds">
    </iframe>
  </iframe>
```

For this setup, if `issuer.example` were to call SPC then origin would be `issuer.example` while the top-level origin would be `merchant.example`. But for this, we care about whether the caller themselves (`issuer.example`) is the owner of the passed-in credentials.

Does that make sense?

Reply:
Yep. I was trying to solve a spec production error. I made this change and it went through! :) Clearly that was the wrong thing to do. However the error comes from using [=origin=]

Reply:
So the problem is that [=origin=] is ambiguous. Check out https://github.com/w3c/secure-payment-confirmation/actions/runs/11894955036/job/33143378569 , it does explain how to fix it:

![image](https://github.com/user-attachments/assets/c3c681d4-a0d5-43ff-8e17-79be8c871421)

So up in our [link-defaults block]](https://github.com/w3c/secure-payment-confirmation/blob/main/spec.bs#L88) we need to add an entry. I think one of the HTML ones is probably the most correct. Let's go with:

```
spec:html; type:dfn; for:/; text:origin
```

?

Reply:
Urgh, the HTML spec makes me sad (see the *new* errors :().

Sorry for the churn, what if we try the *other* suggested HTML default from bikeshed (the environment settings object one above)